Passing configmap with wrong configuration causes nil pointer exception

**Describe the bug**
I was playing with the service with configuration copied from examples, trying to mount configmap as file. I got an error that confused me, and it took me a shameful amount of time to figure out, that I should simply pass the flag "config" instead of "configmap".

**To Reproduce**
Running the service in K8s with the flag `-configmap` without proper preparation causes exceptions which is hard to understand.

**Expected behaviour**
- No exception
- Understandable error log if configmap is unreachable

**Screenshots**
```
time="2022-10-27T16:25:41Z" level=error msg="an empty namespace may not be set when a resource name is provided" app=s3sync-service version=0.3.12
panic: runtime error: invalid memory address or nil pointer dereference
[signal SIGSEGV: segmentation violation code=0x1 addr=0x38 pc=0x15f99d5]

goroutine 1 [running]:
github.com/mazay/s3sync-service/service.(*Config).setDefaults(...)
	/go/src/github.com/mazay/s3sync-service/service/config.go:63
github.com/mazay/s3sync-service/service.readConfigString({0x0?, 0xc000442300?})
	/go/src/github.com/mazay/s3sync-service/service/config.go:175 +0x55
github.com/mazay/s3sync-service/service.getConfig()
	/go/src/github.com/mazay/s3sync-service/service/config.go:140 +0x4d
github.com/mazay/s3sync-service/service.Start()
	/go/src/github.com/mazay/s3sync-service/service/main.go:126 +0x34a
main.main()
	/go/src/github.com/mazay/s3sync-service/main.go:24 +0x17
```

**Environment (please complete the following information):**
 - OS: [Docker in K8s, quay image]
 - Version [e.g. 0.3.12]

--- a/service/main.go
+++ b/service/main.go
@@ -20,10 +20,12 @@ package service
 
 import (
 	"flag"
+	"fmt"
 	"os"
 	"os/signal"
 	"reflect"
 	"strconv"
+	"strings"
 	"sync"
 	"syscall"
 	"time"
@@ -115,6 +117,14 @@ func Start() {
 	flag.Parse()
 
 	status = "STARTING"
+
+	if configmap != "" {
+		err := cmValidate(configmap)
+		if err != nil {
+			logger.Error(err)
+			osExit(7)
+		}
+	}
 
 	if isInK8s() && configmap != "" {
 		clientset = k8sClientset()
@@ -305,4 +315,14 @@ func syncSites(config *Config, uploadCh chan<- UploadCFG,
 			}
 		}(site)
 	}
+}
+
+// cmValidate checks if provided configmap argument matches the pattern
+func cmValidate(cm string) error {
+	cmSplit := strings.Split(cm, "/")
+	if len(cmSplit) != 2 {
+		return fmt.Errorf("unexpected format for the configmap argument, expected format is '<namespace>/<configmap-name>', got '%s'", cm)
+	}
+
+	return nil
 }

--- a/service/main_test.go
+++ b/service/main_test.go
@@ -19,6 +19,8 @@
 package service
 
 import (
+	"errors"
+	"fmt"
 	"os"
 	"testing"
 )
@@ -40,6 +42,24 @@ func TestIsInK8s(t *testing.T) {
 			}
 			if got := isInK8s(); got != tt.want {
 				t.Errorf("isInK8s() = %v, want %v", got, tt.want)
+			}
+		})
+	}
+}
+
+func TestCmValidate(t *testing.T) {
+	tests := []struct {
+		cm     string
+		result bool
+	}{
+		{"configmap-name", false},
+		{"namespace/configmap-name", true},
+	}
+	for _, tt := range tests {
+		t.Run(tt.cm, func(t *testing.T) {
+			if err := cmValidate(tt.cm); errors.Is(err, nil) != tt.result {
+				fmt.Print(errors.Is(err, nil))
+				t.Errorf("cmValidate unexpectedly failed with %s", err)
 			}
 		})
 	}

